lib/index.js

```diff
@@ -1992,6 +1992,47 @@ addMethod('toCallback', function (cb) {
 });
 
 /**
+ * Returns the result of a stream to a nodejs-style callback function.
+ *
+ * If the stream contains a single value, it will call `cb`
+ * with the single item emitted by the stream (if present).
+ * If the stream is empty, `cb` will be called without any arguments.
+ * If an error is encountered in the stream, this function will stop
+ * consumption and call `cb` with the error.
+ * If the stream contains more than one item, it will stop consumption
+ * and call `cb` with an error.
+ *
+ * @id toPromise
+ * @section Consumption
+ * @name Stream.toPromise(Promise)
+ * @param {Function} Promise - Promises/A+ compliant constructor
+ * @api public
+ *
+ * _([1, 2, 3, 4]).collect().toPromise(Promise).then(function (result) {
+ *     // parameter result will be [1,2,3,4]
+ * });
+ */
+
+addMethod('toPromise', function (Promise) {
```

Comment:
Can you name the parameter `PromiseCtor` or something similar to highlight the fact that we do *not* use the global `Promise` object.